app/models/application_type.rb

```diff
@@ -1,4 +1,22 @@
 class ApplicationType < ApplicationRecord
   has_many :applications
   has_many :sources, :through => :applications
+
+  def self.seed
+    seeds = YAML.load_file(Rails.root.join("db/seeds/application_types.yml"))
+
+    transaction do
+      records = all.index_by(&:name)
+
+      seeds.each do |name, attributes|
+        if (r = records.delete(name))
+          r.update!(attributes)
+        else
+          create!(attributes.merge(:name => name))
+        end
+      end
+
+      records.values.map(&:destroy)
```

Comment:
Note I added destroy here for consistency, so if an entry is removed from the file, it will be destroyed...I can remove this if you'd prefer, but I didn't see the harm in adding it for consistency.